application/ui/src/features/inspect/train-model/train-model-dialog.component.tsx

```diff
@@ -38,6 +39,12 @@ export const TrainModelDialog = ({ close }: { close: () => void }) => {
         setSearchParams(searchParams);
     };
     const [selectedModel, setSelectedModel] = useState<string | null>(null);
+    const [selectedDevice, setSelectedDevice] = useState<string | null>(null);
+    const handleDeviceSelect = useCallback((device: string | null) => {
+        setSelectedDevice(device);
+    }, []);
```

Comment:
imho adding the extra `handleDeviceSelect` is not necessary (either rename it here, or use the old name below)

```suggestion
    const [selectedDevice, handleDeviceSelect] = useState<string | null>(null);
```